src/collaborative_platform/apps/dataset_stats/helpers.py

```diff
@@ -0,0 +1,131 @@
+
+    file_contents = tuple((get_content(fv), fv.file.name) for fv in project_version.file_versions.all())
+
+    #files_in_project_version = FileVersion.objects\
+    #    .filter(file__in=project_version.project.files)\
+    #    .filter(creation_date__le=project_version.date)\
+    #    .distinct('file')
+
+    return file_contents        
+
+def create_summary_for_document(doc_raw: str, doc_name:str='undefined')->pd.DataFrame: 
+    doc_tree = et.fromstring(doc_raw.encode())
+
+    header_tags = doc_tree.find('.//tei:teiHeader', namespaces=NAMESPACES).iter()
+    body_tags = doc_tree.find('.//tei:body', namespaces=NAMESPACES).iter()
+
+    stats_df = pd.DataFrame(columns=['document', 'tag', 'tag_id', 'location', 'attr_name', 'attr_value'])
+
+    tag_count = 0
+    for tag in header_tags:
+        for attr_name, attr_value in tag.items():
+            stats_df.loc[len(stats_df)] = {
+                'document': doc_name,
+                'tag': tag.tag,
+                'tag_id': doc_name + str(tag_count),
+                'location': 'header',
+                'attr_name': attr_name,
+                'attr_value': attr_value
+            }
+        tag_count += 1
+        
+    for tag in body_tags:
+        for attr_name, attr_value in tag.items():
+            stats_df.loc[len(stats_df)] = {
+                'document': doc_name,
+                'tag': tag.tag,
+                'tag_id': doc_name + str(tag_count),
+                'location': 'body',
+                'attr_name': attr_name,
+                'attr_value': attr_value
+            }
+        tag_count += 1
+
+    return stats_df
+
+def create_summary_for_document_collection(doc_gen: Iterable[str])->pd.DataFrame: 
+    stats_df = pd.DataFrame(columns=['document', 'tag', 'tag_id', 'location', 'attr_name', 'attr_value'])
+
+    for doc, doc_name in tqdm(doc_gen):
+        stats = create_summary_for_document(doc, doc_name)
+        stats_df = stats_df.append(stats)
+
+    return stats_df
+
+def get_stats(stats_df):
+    n_docs = len(stats_df['document'].unique())
+    tag_g = stats_df.groupby('tag')
+    tag_stats = tag_g.describe()
+    attr_value_counts = stats_df \
+            .loc[:,['tag','attr_name', 'attr_value']] \
+            .groupby(['tag','attr_name']) \
+            .attr_value \
+            .value_counts()
+    
+    stats = tuple({
+        'long_name': row[0],
+        'name': row[0].split('}')[1] if '}' in row[0] else row[0],
+        'count': int(row[1]['tag_id']['unique']),
+        'coverage': float(round(100*row[1]['document']['unique'] / n_docs, 2)),
+        'distinct_doc_occurrences': int(row[1]['document']['unique']),
+        'location': row[1]['location']['top'],
+        'attributes': tuple({
+                'name': attr[0].split('}')[1] if '}' in attr[0] else attr[0],
+                'trend_percentage': float(round(round(100*attr[1]['attr_value']['freq']/attr[1]['attr_value']['count']),2)),
```

Comment:
[click](https://www.youtube.com/watch?v=UfcvO2t8Ntg)

Reply:
I am sorry but it will stay as this is a proof of concept that I am not reinvesting time on.